samples/client/echo_api/dart/dio/built_value/lib/src/api/fake_api.dart

```diff
@@ -0,0 +1,2640 @@
+        
+    
+    Date
+    )
+
+),
+      if (dateTime != null) r'dateTime': await encodeFormParameter(_repository, dateTime, const TypeInfo(
+        
+    
+    DateTime
+    )
+
+),
+      if (password != null) r'password': await encodeFormParameter(_repository, password, const TypeInfo(
+        
+    
+    String
+    )
+
+),
+      if (callback != null) r'callback': await encodeFormParameter(_repository, callback, const TypeInfo(
+        
+    
+    String
+    )
+
+),
+    };
+    _bodyData = _bodyMap;
+
+    final _response = await rawApi.testEndpointParameters(
+      
+      body: _bodyData,
+      requestContentType: 'application/x-www-form-urlencoded',
+      cancelToken: cancelToken,
+      headers: headers,
+      extra: extra,
+      validateStatus: validateStatus,
+      onSendProgress: onSendProgress,
+      onReceiveProgress: onReceiveProgress,
+    );    
+
+    return _response;
+  }
+
+  /// To test enum parameters
+  /// To test enum parameters
+  ///
+  /// Parameters:
+  /// * [enumHeaderStringArray] - Header parameter enum test (string array)
+  /// * [enumHeaderString] - Header parameter enum test (string) (Default: r&#39;-efg&#39;)  
+  /// * [enumQueryStringArray] - Query parameter enum test (string array)
+  /// * [enumQueryString] - Query parameter enum test (string) (Default: r&#39;-efg&#39;)  
+  /// * [enumQueryInteger] - Query parameter enum test (double)
+  /// * [enumQueryDouble] - Query parameter enum test (double)
+  /// * [enumQueryModelArray] 
+  /// * [enumFormStringArray] - Form parameter enum test (string array) (Default: r&#39;$&#39;)  
+  /// * [enumFormString] - Form parameter enum test (string) (Default: r&#39;-efg&#39;)  
+  /// * [cancelToken] - A [CancelToken] that can be used to cancel the operation
+  /// * [headers] - Can be used to add additional headers to the request
+  /// * [extras] - Can be used to add flags to the request
+  /// * [validateStatus] - A [ValidateStatus] callback that can be used to determine request success based on the HTTP status of the response
+  /// * [onSendProgress] - A [ProgressCallback] that can be used to get the send progress
+  /// * [onReceiveProgress] - A [ProgressCallback] that can be used to get the receive progress
+  ///
+  /// Returns a [Future]
+  /// Throws [DioException] if API call or serialization fails
+  Future<Response<void>> testEnumParameters({ 
+    required BuiltList<EnumHeaderStringArrayEnum> enumHeaderStringArray,
+     EnumHeaderStringEnum enumHeaderString = r'-efg',
+    required BuiltList<EnumQueryStringArrayEnum> enumQueryStringArray,
+     EnumQueryStringEnum enumQueryString = r'-efg',
+    required EnumQueryIntegerEnum enumQueryInteger,
+    required EnumQueryDoubleEnum enumQueryDouble,
+    required BuiltList<ModelEnumClass> enumQueryModelArray,
+     BuiltList<InnerEnum> enumFormStringArray = r'$',
+     EnumFormStringEnum enumFormString = r'-efg',
```

Comment:
Need to find a way to generate these defaults correctly.
And also collection default values.

Reply:
with `built_value`, you can't since you need const constructors, so I plan on removing support for `built_value` entirely and focus on `json_serializable`/`freezed`